Formula/flagd.rb

```diff
@@ -0,0 +1,20 @@
+class Flagd < Formula
+  desc "Feature flag daemon with a Unix philosophy"
+  homepage "https://github.com/open-feature/flagd"
+  url "https://github.com/open-feature/flagd.git", using: :git, tag: "v0.3.0", revision: "2ad206e635cc5b68ae42287b740ddfc86b0194ff"
+  license "Apache-2.0"
+  head "https://github.com/open-feature/flagd.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPRIVATE"] = "buf.build/gen/go"
+    system "go", "build", "-o", bin/"flagd"
+  end
+
+  test do
+    assert_predicate bin/"flagd", :exist?
+
+    assert_match(/Flagd is a simple command line tool/, shell_output("#{bin}/flagd"))
```

Comment:
We need to assert the command usage, not help text.

Reply:
@chenrui333 Since this is a daemon process, do we have any examples or testing services that I can use as reference?

Reply:
HTTPD is a daemon process that is tested.

Reply:
also as this is daemon process, you can add plist as well

Reply:
A service block, not a plist.

Reply:
yeah, service block (we deprecated plist) :)

Reply:
Thanks @SMillerDev and @chenrui333 . I've added tests for this